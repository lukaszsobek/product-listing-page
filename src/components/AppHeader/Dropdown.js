import React from 'react'

const Dropdown = ({
  clearFilter, data, dropdownClass, activeFilters, isOpen, toggleFilter
}) => {
  if (!isOpen) {
    return null
  }

  const activeFilterString = activeFilters.join(',,,')

  const list = data.map((dataItem, key) => {
    const isFilterActive = activeFilterString.indexOf(dataItem) > -1

    return (
      <div
        className={isFilterActive ? 'active' : ''}
        key={key}
        onClick={() => toggleFilter(dataItem)}
      >{dataItem}</div>
    )
  })

  return (
    <div className={dropdownClass}>
      { list }
      { activeFilters.length > 0 &&
        <div
          onClick={clearFilter}
        >...clear</div>
      }
    </div>
  )
}

export default Dropdown
