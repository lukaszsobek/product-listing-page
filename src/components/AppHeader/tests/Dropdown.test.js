import Dropdown from "../Dropdown";

describe("DropDown", () => {

    it("returns null if isOpen is false",() => {
        const props = {}
        expect(Dropdown(props)).toBe(null);
    });

    it("renders DropDown",() => {
        const props = {
            isOpen: true,
            data: [1,2],
            dropdownClass: "testClass"

        }
        expect(Dropdown(props)).toMatchSnapshot();
    });

});