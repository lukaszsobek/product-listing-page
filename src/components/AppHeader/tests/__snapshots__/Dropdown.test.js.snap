// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown renders DropDown 1`] = `
<div
  className="testClass"
>
  <div
    onClick={[Function]}
  >
    1
  </div>
  <div
    onClick={[Function]}
  >
    2
  </div>
</div>
`;
